docs/i18n.md

```diff
@@ -0,0 +1,86 @@
+# Interna... Internazio... Internationaliz... I18n.
+
+Cruddl provides a sophisticated way to provide human readable field and type names and storing them along where they belong: together with the model. A client consuming a cruddl graphql api can not only query data and meta schema, but also fetch labels and hints. 
+
+## How to define localization
+A cruddl project contains graphql schema files and json/yaml object files. All localization entries must be located an object `i18n`. Inside of the `i18n` object, <i>namespaces</i> can be defined beginning at the root namespace:  
+```yaml
+i18n:
+  en:
+    types:
+      Hero:
+        singular: Hero
+        plural: Heroes
+        hint: A brave person with cool features
+        fields:
+          name:
+            label: Name
+            hint: The name of the hero
+          age: age # shortcut for age: { label: age, hint: undefined }
+    fields:
+      createdAt:
+        label: Created at
+        hint: When this hero's record was created
+  de:
+    types:
+      Hero:
+        ...
+    fields:
+      createdAt:
+        label: Erfasst am
+        hint: Wann der Held in diesem System angelegt wurde
+      updatedAt: Geändert am
+      id: ID
+    namespaces:
```

Comment:
I'm not really sure if we need this `namespaces` section here. Limiting field and type translations to a namespace should be done via the `namespacePath` on the `ProjectSource`, i.e. by the directory of the yaml file.

Reply:
Providing namespaces allows you to do all I18n-stuff with only one file. I think it's ok.

Reply:
ok, having it won't hurt